packages/strategies/stress.go

```diff
@@ -70,65 +87,220 @@ func (s *StressTest) modifyTransactions(input *types.Block, transactions types.T
 
 	result := types.Transactions{}
 
-	depositTxns := types.Transactions{}
-	userTxns := types.Transactions{}
-
-	for _, txn := range transactions {
-		if txn.IsDepositTx() {
-			depositTxns = append(depositTxns, txn)
-		} else {
-			userTxns = append(userTxns, txn)
-		}
-	}
+	depositTxns, userTxns := s.splitTxs(transactions)
 
 	currentBlockNum := input.NumberU64()
 
 	startBlockNum := s.startBlock.NumberU64()
 	// If first 10 blocks: Add one deposit transactions to send ETH to accounts
 	// Maybe if seqNum=0 (?)
 	if currentBlockNum < startBlockNum+11 {
-		i := currentBlockNum - startBlockNum - 1
-		addr := addresses[i]
+		dep := s.prepareDeposit(l1Info, currentBlockNum)
+		depositTxns = append(depositTxns, dep)
 
-		// Add deposit transaction
-		var dep types.DepositTx
+		if s.cfg.InjectERC20 && currentBlockNum == startBlockNum+1 {
+			erc20, err := s.prepareDeployErc20(input)
+			if err != nil {
+				panic(err)
+			}
 
-		source := derive.UserDepositSource{
-			L1BlockHash: l1Info.BlockHash, // l1 block info
-			LogIndex:    10_000,           // hardcode to a very large number
+			userTxns = append(types.Transactions{erc20}, userTxns...)
 		}
+	} else {
+		if s.cfg.InjectERC20 {
+			if erc20Address == (common.Address{}) {
+				s.waitForFirstBlock()
```

Comment:
If we reach this part of the code, we have already gone through 10 blocks above for deposit txs (in the `if` block). Do we need to wait for another block here? Or could we get rid of `s.waitForFirstBlock`?

Reply:
Waiting for blocks through a channel to get the contract address is a dumb way to do it. Just ignore that; I've made the changes needed.

Now I will use the `crypto.CreateAddress` method to get the erc20 contract address.